client_test.go

```diff
@@ -356,6 +356,70 @@ func TestCreateDatasetThatAlreadyExists(t *testing.T) {
 	assert.Equal(errVolumeAlreadyExists, err)
 }
 
+func TestCreateDatasetThatFailsServerSide(t *testing.T) {
+	const (
+		datasetName       = "dir"
+		expectedPrimary   = "A-B-C-D"
+		expectedDatasetID = "uuid-1"
+	)
+
+	// reduce timeouts for testing
+	tickerWaitingForVolume = 1 * time.Microsecond
+	timeoutWaitingForVolume = 1 * time.Millisecond
+
+	assert := assert.New(t)
+	var numCalls int
+	var deleted bool
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		numCalls++
+		switch numCalls {
+		case 1:
+			assert.Equal("GET", r.Method)
+			assert.Equal("/v1/state/nodes", r.URL.Path)
+			w.Write([]byte(fmt.Sprintf(`[{"host": "127.0.0.1", "uuid": "%s"}]`, expectedPrimary)))
+		case 2:
+			assert.Equal("POST", r.Method)
+			assert.Equal("/v1/configuration/datasets", r.URL.Path)
+			w.Write([]byte(fmt.Sprintf(`{"dataset_id": "%s"}`, expectedDatasetID)))
+		default:
+			if r.Method == "GET" {
+				assert.Equal("/v1/state/datasets", r.URL.Path)
+				w.Write([]byte(`[]`))
```

Comment:
:question: Yeah, I don't think this is exercising the GetDatasetState error cleanup code...only the timeout situation. 
Another reason why I'd be inclined to only attempt the cleanup when dataset creation reaches a timeout.